der/tests/derive.rs

```diff
@@ -261,6 +353,60 @@ mod sequence {
     const ALGORITHM_IDENTIFIER_DER: &[u8] =
         &hex!("30 13 06 07 2a 86 48 ce 3d 02 01 06 08 2a 86 48 ce 3d 03 01 07");
 
+    #[test]
+    fn idp_test() {
+        let idp = IssuingDistributionPointExample::from_der(&hex!("30038101FF")).unwrap();
+        assert_eq!(idp.only_contains_user_certs, true);
+        assert_eq!(idp.only_contains_cacerts, false);
+        assert_eq!(idp.indirect_crl, false);
+        assert_eq!(idp.only_contains_attribute_certs, false);
+
+        let idp = IssuingDistributionPointExample::from_der(&hex!("30038201FF")).unwrap();
+        assert_eq!(idp.only_contains_user_certs, false);
+        assert_eq!(idp.only_contains_cacerts, true);
+        assert_eq!(idp.indirect_crl, false);
+        assert_eq!(idp.only_contains_attribute_certs, false);
+
+        let idp = IssuingDistributionPointExample::from_der(&hex!("30038401FF")).unwrap();
+        assert_eq!(idp.only_contains_user_certs, false);
+        assert_eq!(idp.only_contains_cacerts, false);
+        assert_eq!(idp.indirect_crl, true);
+        assert_eq!(idp.only_contains_attribute_certs, false);
+
+        let idp = IssuingDistributionPointExample::from_der(&hex!("30038501FF")).unwrap();
+        assert_eq!(idp.only_contains_user_certs, false);
+        assert_eq!(idp.only_contains_cacerts, false);
+        assert_eq!(idp.indirect_crl, false);
+        assert_eq!(idp.only_contains_attribute_certs, true);
+    }
+
+    // demonstrates default field that is not context specific
+    #[test]
+    fn extension_test() {
+        let ext1 = ExtensionExample::from_der(&hex!(
+            "
+        300F        //  0  15: SEQUENCE {
+        0603551D13  //  2   3:   OBJECT IDENTIFIER basicConstraints (2 5 29 19)
+        0101FF      //  7   1:   BOOLEAN TRUE
+        0405        //  10   5:   OCTET STRING, encapsulates {
+        3003        //  12   3:     SEQUENCE {
+        0101FF      //  14   1:       BOOLEAN TRUE
+        "
+        ))
+        .unwrap();
+        assert_eq!(ext1.critical, true);
+
+        let ext2 = ExtensionExample::from_der(&hex!("
+        301F                                            //  0  31: SEQUENCE {
+        0603551D23                                      //  2   3:   OBJECT IDENTIFIER authorityKeyIdentifier (2 5 29 35)
+        0418                                            //  7  24:   OCTET STRING, encapsulates {
+        3016                                            //  9  22:     SEQUENCE {
+        8014E47D5FD15C9586082C05AEBE75B665A7D95DA866    // 11  20:       [0] E4 7D 5F D1 5C 95 86 08 2C 05 AE BE 75 B6 65 A7 D9 5D A8 66
+        "))
```

Comment:
Nice! But maybe indent these consistently?

Reply:
They both use minimum tab stop for the left hand code. I can bump the first example out to match the second if you like.

Reply:
Perhaps indented like so?
```suggestion
            "
            300F        //  0  15: SEQUENCE {
            0603551D13  //  2   3:   OBJECT IDENTIFIER basicConstraints (2 5 29 19)
            0101FF      //  7   1:   BOOLEAN TRUE
            0405        //  10   5:   OCTET STRING, encapsulates {
            3003        //  12   3:     SEQUENCE {
            0101FF      //  14   1:       BOOLEAN TRUE
            "
        ))
        .unwrap();
        assert_eq!(ext1.critical, true);

        let ext2 = ExtensionExample::from_der(&hex!(
            "
            301F                                            //  0  31: SEQUENCE {
            0603551D23                                      //  2   3:   OBJECT IDENTIFIER authorityKeyIdentifier (2 5 29 35)
            0418                                            //  7  24:   OCTET STRING, encapsulates {
            3016                                            //  9  22:     SEQUENCE {
            8014E47D5FD15C9586082C05AEBE75B665A7D95DA866    // 11  20:       [0] E4 7D 5F D1 5C 95 86 08 2C 05 AE BE 75 B6 65 A7 D9 5D A8 66
            "
        ))
```

Not sure if rustfmt will be happy with that, but...